lib/core.js

```diff
@@ -539,6 +543,20 @@ module.exports = class Core {
     return verifies[0] !== null
   }
 
+  async flushHints() {
```

Comment:
ATM there are race conditions possible. It's trivially solved by debouncing this function and setting `this.hintsChanged = false` right after the early-return check (in the same tick).

Reply:
fixed :)

Reply:
Clarified out of band that this runs in a lock, so we are fully protected from race conditions by setting `hintsChanged = false` in the same tick